incubator/bitcoind/templates/deployment.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "bitcoind.fullname" . }}
+  labels:
+    app: {{ template "bitcoind.name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  template:
+    metadata:
+      labels:
+        app: {{ template "bitcoind.name" . }}
+        release: "{{ .Release.Name }}"
+    spec:
+      containers:
+        - name: {{ template "bitcoind.fullname" . }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          resources:
+  {{ toYaml .Values.resources | indent 10 }}
+          ports:
+            - name: rpc
+              containerPort: {{ .Values.rpcPort }}
+          volumeMounts:
+            - name: data
+              mountPath: /root
+            {{- if .Values.ssl.enabled }}
+            - name: ssl
+              mountPath: /etc/bitcoind-ssl
+            {{- end }}
+            {{- if .Values.configurationFiles }}
+            - name: configurations
+              mountPath: /etc/bitcoind
+            {{- end }}
+      volumes:
+        {{- if .Values.ssl.enabled }}
```

Comment:
Why are we enabling SSL at the chart level and not relying on an ingress? Ingress seems to be the dominant pattern.